R/tbl_summary.R

```diff
@@ -233,8 +233,87 @@ tbl_summary <- function(data, by = NULL, label = NULL, statistic = NULL,
   tbl_summary_inputs <- as.list(environment())
 
   # removing variables with unsupported variable types from data ---------------
-  data <- select(data, !!include)
   classes_expected <- c("character", "factor", "numeric", "logical", "integer", "difftime")
+  data <- removing_variables_with_unsupported_types(data, include, classes_expected)
+
+  # checking function inputs ---------------------------------------------------
+  tbl_summary_input_checks(
+    data, by, label, type, value, statistic,
+    digits, missing, missing_text, sort
+  )
+
+  # generate meta_data --------------------------------------------------------
+  meta_data <- generate_metadata(data, value, by, classes_expected, type, label, statistic, digits, percent, sort)
+
+  # calculating summary statistics ---------------------------------------------
+  table_body <-
+    meta_data %>%
+    mutate(
+      tbl_stats = pmap(
+        list(.data$summary_type, .data$variable,
+             .data$var_label, .data$stat_display, .data$df_stats),
+        function(summary_type, variable, var_label, stat_display, df_stats) {
+          df_stats_to_tbl(
+            data = data, variable = variable, summary_type = summary_type, by = by,
+            var_label = var_label, stat_display = stat_display,
+            df_stats = df_stats, missing = missing, missing_text = missing_text
+          )
+        }
+      )
+    ) %>%
+    pull(.data$tbl_stats) %>%
+    purrr::reduce(bind_rows)
+
+  # table of column headers ----------------------------------------------------
+  table_header <-
+    tibble(column = names(table_body)) %>%
+    table_header_fill_missing() %>%
+    mutate(
+      # adding footnote of statistics on display (unless theme indicates a no print)
+      footnote = ifelse(
+        startsWith(.data$column, "stat_"),
+        footnote_stat_label(meta_data),
+        .data$footnote
+      )
+    )
+
+  # returning all results in a list --------------------------------------------
+  results <- list(
+    table_body = table_body,
+    table_header = table_header,
+    meta_data = meta_data,
+    inputs = tbl_summary_inputs,
+    N = nrow(data),
+    call_list = list(tbl_summary = match.call())
+  )
+  results$by <- by
+  results$df_by <- df_by(data, by)
+
+  # assigning a class of tbl_summary (for special printing in Rmarkdown)
+  class(results) <- c("tbl_summary", "gtsummary")
+
+  # adding headers
+  if (is.null(by)) {
+    results <- modify_header_internal(
+      results,
+      stat_0 = "**N = {N}**",
+      label = paste0("**", translate_text("Characteristic"), "**")
+    )
+  } else {
+    results <- modify_header_internal(
+      results,
+      stat_by = "**{level}**, N = {n}",
+      label = paste0("**", translate_text("Characteristic"), "**")
+    )
+  }
+
+  return(results)
+}
+
+
+# removing variables with unsupported variable types from data -------------------------
+removing_variables_with_unsupported_types <- function(data, include, classes_expected) {
```

Comment:
According to labelled's philosophy, labelled vectors should not produce an error but treated as is they were unclassed (i.e. a labelled double to be treated as continuous and a labelled character as categorical). 

However, it could be relevant to display a message to indicate that a labelled vector was found and treated as continuous/categorical.

Regards

Reply:
I would like to address this issue in another pull request. https://github.com/ddsjoberg/gtsummary/issues/569

I eventually want to update to accept all columns regardless of their class (base and non-base R classes).